Tests/SystemTests/NetworkConnectivityTests/Subscriber/Helper/SubscriberMonitorFactory.swift

```diff
@@ -0,0 +1,319 @@
+import Foundation
+import AblyAssetTrackingInternal
+import AblyAssetTrackingCore
+import AblyAssetTrackingSubscriber
+import AblyAssetTrackingSubscriberTesting
+import struct Combine.AnyPublisher
+
+extension SubscriberNetworkConnectivityTests {
+    class SubscriberMonitorFactory {
+        private let sharedState: SubscriberMonitor.SharedState
+        private let subscriber: Subscriber
+        private let logHandler: InternalLogHandler
+        private let trackableID: String
+        private let faultType: FaultTypeDTO
+        private let subscriberClientID: String
+        private let subscriberResolutionPreferences: AnyPublisher<Resolution, Never>
+
+        init(subscriber: Subscriber, combineSubscriberDelegate: CombineSubscriberDelegate, logHandler: InternalLogHandler, trackableID: String, faultType: FaultTypeDTO, subscriberClientID: String, subscriberResolutionPreferences: AnyPublisher<Resolution, Never>) {
+            self.sharedState = .init(combineSubscriberDelegate: combineSubscriberDelegate)
+            self.subscriber = subscriber
+            self.logHandler = logHandler
+            self.trackableID = trackableID
+            self.faultType = faultType
+            self.subscriberClientID = subscriberClientID
+            self.subscriberResolutionPreferences = subscriberResolutionPreferences
+        }
+
+        /**
+         * Construct `SubscriberMonitor` configured to expect appropriate state transitions
+         * for the given fault type while it is active. `label` will be used for logging captured transitions.
+         */
+        func forActiveFault(
+            label: String,
+            locationUpdate: Location? = nil,
+            publisherResolution: Resolution? = nil,
+            publisherDisconnected: Bool = false,
+            subscriberResolution: Resolution? = nil
+        ) -> SubscriberMonitor {
+            let expectedState: ConnectionState
+            switch faultType {
+            case .fatal:
+                expectedState = .failed
+            case .nonfatal where publisherDisconnected:
+                expectedState = .offline
+            case .nonfatal:
+                expectedState = .online
+            case .nonfatalWhenResolved:
+                expectedState = .offline
+            }
+
+            let failureStates: Set<ConnectionState>
+            switch faultType {
+            case .fatal:
+                failureStates = [.offline]
```

Comment:
Seems like it would be. It's a bit of a moot point right now since there are no fatal tests anyway. Like I said, I just copied the assertions from Android ([here's this one](https://github.com/ably/ably-asset-tracking-android/blob/f6d163fc47aa329e6f7e41d5cb4a15ee87c4bc43/subscribing-sdk/src/androidTest/java/com/ably/tracking/subscriber/NetworkConnectivityTests.kt#L645)) without applying much of a critical eye. Reckon I should change it here or keep it consistent with Android?

Reply:
I think its adding here and then creating an issue on the Android side to bring parity